examples/lump-mpi.py

```diff
@@ -131,21 +129,68 @@ def my_rhs(t, state):
                               boundaries=boundaries, eos=eos)
 
     def my_checkpoint(step, t, dt, state):
-        return sim_checkpoint(discr, visualizer, eos, cv=state,
-                              exact_soln=initializer, vizname=casename, step=step,
-                              t=t, dt=dt, nstatus=nstatus, nviz=nviz,
-                              exittol=exittol, constant_cfl=constant_cfl, comm=comm)
-
-    try:
-        (current_step, current_t, current_state) = \
-            advance_state(rhs=my_rhs, timestepper=timestepper,
-                checkpoint=my_checkpoint,
-                get_timestep=get_timestep, state=current_state,
-                t=current_t, t_final=t_final)
-    except ExactSolutionMismatch as ex:
-        current_step = ex.step
-        current_t = ex.t
-        current_state = ex.state
+        from mirgecom.simutil import check_step
+        do_status = check_step(step=step, interval=nstatus)
+        do_viz = check_step(step=step, interval=nviz)
+        do_health = check_step(step=step, interval=nhealth)
+
+        if do_status or do_viz or do_health:
+            from mirgecom.simutil import compare_fluid_solutions
+            dv = eos.dependent_vars(state)
+            exact_mix = initializer(x_vec=nodes, eos=eos, t=t)
+            component_errors = compare_fluid_solutions(discr, state, exact_mix)
+            resid = state - exact_mix
+            io_fields = [
+                ("cv", state),
+                ("dv", dv),
+                ("exact_mix", exact_mix),
+                ("resid", resid)
+            ]
+
+        if do_status:  # This is bad, logging already completely replaces this
+            from mirgecom.io import make_status_message
+            status_msg = make_status_message(discr=discr, t=t, step=step, dt=dt,
+                                             cfl=current_cfl, dependent_vars=dv)
+            status_msg += (
+                "\n------- errors="
+                + ", ".join("%.3g" % en for en in component_errors))
+            if rank == 0:
+                logger.info(status_msg)
+
+        errored = False
+        if do_health:
+            from mirgecom.simutil import check_naninf_local, check_range_local
+            if check_naninf_local(discr, "vol", dv.pressure) \
+               or check_range_local(discr, "vol", dv.pressure, .9, 1.1):
+                errored = True
+                message = "Invalid pressure data found.\n"
+            exittol = .09
+            if max(component_errors) > exittol:
+                errored = True
+                message += "Solution diverged from exact_mix.\n"
+            comm = discr.mpi_communicator
+            if comm is not None:
+                errored = comm.allreduce(errored, op=MPI.LOR)
+            if errored:
+                if rank == 0:
+                    logger.info("Fluid solution failed health check.")
+                logger.info(message)   # do this on all ranks
+
+        if do_viz or errored > 0:
```

Comment:
How about this?
```suggestion
        if do_viz or errored:
```
(similarly for other instances of this in the other examples)

Reply:
I did not resolve this because maybe you should have a quick peak at it again.  I updated this to be like `autoignition`.